### PR TITLE
8258058: improve description of OutOfMemoryError relevant flags

### PR DESCRIPTION
**Pull request by navyxliu**
/cc hotspot

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8258058](https://bugs.openjdk.java.net/browse/JDK-8258058): improve description of OutOfMemoryError relevant flags

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Jie Fu](https://openjdk.java.net/census#jiefu) (@DamonFool - Committer)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1792/head:pull/1792`
`$ git checkout pull/1792`

### CODE REVIEW

**Comment by navyxliu** [comment]
/summary OutOfMemoryError thrown out of JVM doesn't notify hotspot, so debug.cpp::report_java_out_out_memory does not handle it. ie. Some options don't respect OOMEs from Java libraries.  We need to make the description more clear to users.

**Review by dholmes-ora — APPROVED** [review]
Seems fine.

Thanks,
David

File: src/hotspot/share/runtime/globals.hpp [hunk 1]
```diff
@@ -550,7 +551,8 @@ const intx ObjectAlignmentInBytes = 8;
           "Dump heap to file after any major stop-the-world GC")            \
                                                                             \
   product(bool, HeapDumpOnOutOfMemoryError, false, MANAGEABLE,              \
-          "Dump heap to file when java.lang.OutOfMemoryError is thrown")    \
+          "Dump heap to file when java.lang.OutOfMemoryError is thrown "    \
+          "from JVM")    \
```
Comment (DamonFool) [inline src/hotspot/share/runtime/globals.hpp:555]: It would be better to align `\` to the right.

Comment (navyxliu) [reply src/hotspot/share/runtime/globals.hpp:555]: yes, thanks!

**Review by DamonFool — APPROVED** [review]
Thanks for the change.

LGTM

**Review by dholmes-ora — APPROVED** [review]

**Comment by navyxliu** [comment]
Thanks! @DamonFool  and @dholmes-ora

**Pull request closed without merging**